src/main/groovy/life/qbic/auth/Authentication.groovy

```diff
@@ -0,0 +1,74 @@
+package life.qbic.auth
+
+import groovy.util.logging.Log4j2
+import io.micronaut.context.annotation.Value
+import io.micronaut.core.annotation.Nullable
+import io.micronaut.http.HttpRequest
+import io.micronaut.security.authentication.AuthenticationException
+import io.micronaut.security.authentication.AuthenticationFailed
+import io.micronaut.security.authentication.AuthenticationProvider
+import io.micronaut.security.authentication.AuthenticationRequest
+import io.micronaut.security.authentication.AuthenticationResponse
+import io.micronaut.security.authentication.UserDetails
+import io.reactivex.Flowable
+import org.reactivestreams.Publisher
+import org.yaml.snakeyaml.Yaml
+
+import javax.annotation.PostConstruct
+import javax.inject.Singleton
+
+@Log4j2
+@Singleton
+class Authentication implements AuthenticationProvider{
+
+    @Value('${userroles.config}')
+    String configPath
+
+    private Map config
+
+    @PostConstruct
+    void initialize() {
```

Comment:
Here a bit of JavaDoc would be helpful to understand what is intended to happen in this method.